R/plot_model_profile_survival.R

```diff
@@ -32,89 +36,415 @@
+            x_width = x_width,
+            plot_type = plot_type,
+            single_timepoint = single_timepoint,
+            colors = color_scale
+        )
+    }
+
+    pl +
+        labs(
+            title = title,
+            subtitle = subtitle
+        ) +
+        theme_default_survex()
+}
+
+plot_pdp_num <- function(pdp_dt,
+                         ice_dt,
+                         data_dt,
+                         feature_name_sym,
+                         y_floor_ice,
+                         y_ceiling_ice,
+                         y_floor_pd,
+                         y_ceiling_pd,
+                         x_width,
+                         plot_type,
+                         single_timepoint,
+                         colors) {
+    with(pdp_dt, { # to get rid of Note: no visible binding for global variable ...
+        if (single_timepoint == TRUE) { ## single timepoint
+            if (plot_type == "ice") {
+                ggplot(data = ice_dt, aes(x = !!feature_name_sym, y = predictions)) +
+                    geom_line(alpha = 0.2, mapping = aes(group = id), color = colors_discrete_drwhy(1)) +
+                    geom_rug(data = data_dt, aes(x = !!feature_name_sym, y = y_ceiling_pd), sides = "b", alpha = 0.8, position = position_jitter(width=0.01 * x_width)) +
+                    ylim(y_floor_ice, y_ceiling_ice)
+            }
+            # PDP + ICE
+            else if (plot_type == "pdp+ice") {
+                ggplot(data = ice_dt, aes(x = !!feature_name_sym, y = predictions)) +
+                    geom_line(mapping = aes(group = id), alpha = 0.2) +
+                    geom_line(data = pdp_dt, aes(x = !!feature_name_sym, y = pd), linewidth = 2, color = colors_discrete_drwhy(1)) +
+                    geom_rug(data = data_dt, aes(x = !!feature_name_sym, y = y_ceiling_pd), sides = "b", alpha = 0.8, position = position_jitter(width=0.01 * x_width)) +
+                    ylim(y_floor_ice, y_ceiling_ice)
+            }
+            # PDP
+            else if (plot_type == "pdp" || plot_type == "ale") {
+                ggplot(data = pdp_dt, aes(x = !!feature_name_sym, y = pd)) +
+                    geom_line(color = colors_discrete_drwhy(1)) +
+                    geom_rug(data = data_dt, aes(x = !!feature_name_sym, y = y_ceiling_pd), sides = "b", alpha = 0.8, position = position_jitter(width=0.01 * x_width)) +
+                    ylim(y_floor_pd, y_ceiling_pd)
+            }
+        } else { ## multiple timepoints
+            if (plot_type == "ice") {
+                ggplot(data = ice_dt, aes(x = !!feature_name_sym, y = predictions)) +
+                    geom_line(alpha = 0.2, mapping = aes(group = interaction(id, time), color = time)) +
+                    geom_rug(data = data_dt, aes(x = !!feature_name_sym, y = y_ceiling_ice), sides = "b", alpha = 0.8, position = position_jitter(width=0.01 * x_width)) +
+                    scale_color_manual(name = "time", values = colors) +
```

Comment:
Would it maybe be nicer here to have a color scale to display different time values, so it is equivalent to plot(exp, geom = "time", variables = c(“some_numeric_variable”))?

Reply:
Not sure if I understand correctly, but for example, when using `plot(exp, geom = "variable", variables = "some_numeric_variable", times = c(1, 5, 10, 20))` there's a color scale to display different time points. However, time points must be given in `times`, otherwise only one is chosen - the median of the times.

Reply:
I forgot to mention that I was referring to the plot's legend, not the plot itself, sorry. Attached an example screenshot for reference. 
<img width="916" alt="Bildschirmfoto 2023-08-21 um 09 57 53" src="https://github.com/ModelOriented/survex/assets/79644349/a5424311-c462-45a0-b394-c75c4e75b0d7">

Reply:
Ah, yes, that definitely needed a fix 😅 - already changed

Reply:
ahh, I didn't notice that, should be fixed now 😀

but what do you think about making the color scale associated with time continuous or discrete based on the number of time points selected, not always continuous?